### DISCUSSION TITLE
Cloudflared Security

### DISCUSSION THREAD

**Discussion by Orgbrat** [homeassistant-apps/app-cloudflared#470] [Q&A] [answered]
I am using the Home Assistant Cloudflared Addon with the Cloudflare Tunnel which I really like. But I am receiving 1-2 failed login attampts daily. I believe they are only failed because I have 2FA enabled on my Home Assistant instance. I thought the use of the Cloudflare tunnel would have controlled these unwanted login attempts. What do I need to do to thwart off these login attempts.

I really thought there would be a login path into the Cloudflare tunnel and then on to my  login of my Home Assistant instance.
Have a missed  setting something up on Cloudflare's side to stop this login attempts earlier in the access process? If so can someone tell me what I need to do to secure this process better.

**Comment by Orgbrat** [comment, marked as answer]
Well my solution for now was to add the authorization process described here :  [Restrict Access To Cloudflare Tunnel: What You Should Know](https://www.youtube.com/watch?v=cahInL0qg-Q)  Not sure if this will be my permanent solution but it will at least make it harder for whomever is trying to reach my Home Assistant instance for the time being. 

I am open to further suggestions though.

**Comment by brenner-tobias** [comment]
Your way was exactly right: Create an [Access Application in Cloudflare ZeroTrust Dashboard](https://developers.cloudflare.com/cloudflare-one/applications/configure-apps/self-hosted-apps/) and add another layer of login security, including 2FA.
